lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/schema/OrganizationDaoImpl.java

```diff
@@ -80,10 +82,16 @@ public Organization getOrganizationByName(String name) {
 	@Override
 	public void deleteOrganization(String id) {
 		ValidateArgument.required(id, "id");
-		int count = jdbcTemplate.update("DELETE FROM " + TABLE_ORGANIZATION + " WHERE " + COL_ORGANIZATION_ID + " = ?",
-				id);
-		if (count < 1) {
-			throw new NotFoundException("Organization with id: '" + id + "' not found");
+		try {
+			int count = jdbcTemplate.update("DELETE FROM " + TABLE_ORGANIZATION + " WHERE " + COL_ORGANIZATION_ID + " = ?",
+					id);
+			if (count < 1) {
+				throw new NotFoundException("Organization with id: '" + id + "' not found");
+			}
+		} catch (DataIntegrityViolationException e) {
```

Comment:
This might be too much of an umbrella for that specific message that might hide other issues.